### ISSUE TITLE
Resource Landing Page Edit Mode: Required Metadata Message Persists Too Long

### ISSUE THREAD

**Issue by horsburgh** [hydroshare/hydroshare#1411]
When I first create a resource, I am taken to the resource landing page and I get a green message at the top listing any required metadata elements that I am missing. Then, I go into edit mode.  The message at the top of the page is the same, but it turns yellow.  Then, I can specify all of the required elements in edit mode, click their respective save buttons, and I even get the green message that pops up saying my metadata are sufficient to share my resource or make it public, but the original yellow message at the top of the page stating that there are still required elements doesn't go away until I leave edit mode.  

I think this message should probably be modified either: 1) each time they save an individual metadata element, or 2) it should remain until they have specified all of the required elements and should disappear at the same time the green message pops up stating that the metadata are sufficient for sharing. Otherwise the user could be confused about why the yellow message is still at the top of the page even though they have created all of the required elements.

**Comment by AmberSJones** [comment]
I have experienced a related issue. I am editing an existing resource and adding/removing subject keywords. Every time I make a change, the green message appears telling me that the metadata are sufficient to share/make public. I don't need to see the message in this case nor with this frequency. Let me know if this should be posted as a separate issue.

**Comment by pkdash** [comment]
@horsburgh Can you test on beta and see if this got fixed then close it?

**Comment by horsburgh** [comment]
I just tested on beta.hydroshare.org and this issue remains. I did the following:

1. Created a new generic resource with a title.
2. Entered edit mode
3. Created an abstract
4. Created a keyword (I get the green message stating my metadata are sufficient)

The yellow message telling me that I need an Abstract and Keywords is still shown at the top of the resource. It doesn't go away until I leave edit mode.

On a related note - I clicked the button to remove the one keyword I added and get the following error:

![image](https://cloud.githubusercontent.com/assets/3485902/20997197/96ddce84-bcc0-11e6-9dd6-1fb3a120b3b1.png)

This error seems a little misleading as it shouldn't really be an error. It should probably just be a message stating that the resource cannot be made public or published without at least one keyword.  And, it fails to actually remove the last keyword. This might have to be a new issue.

The issue referenced by @AmberSJones (see issue #484) also remains.
